scripting/chat-processor.sp

```diff
@@ -279,14 +294,16 @@ public void Frame_OnChatMessage_SayText2(any data)
 
 	if (iResults == Plugin_Changed)
 	{
-		for (int i = 0; i < GetArraySize(hRecipients); i++)
-		{
-			int client = GetArrayCell(hRecipients, i);
-
-			if (IsClientInGame(client))
-			{
-				CSayText2(client, sBuffer, iSender, bChat);
-			}
+		if(!bProto) {
+			Handle hMsg = StartMessage("SayText2", iRecipients, iNumRecipients, USERMSG_RELIABLE|USERMSG_BLOCKHOOKS);
+			
+			BfWriteByte(hMsg, iSender);
+			BfWriteByte(hMsg, bChat);
+			BfWriteString(hMsg, sBuffer);
+			EndMessage();
+		
+		} else {
+			PbSetString(bfMsg, "msg_name", sBuffer);
```

Comment:
This should not be needed as we are modifying the message only, so the other keys should be left untouched as they originally were.